uaclient/http/__init__.py

```diff
@@ -99,17 +108,42 @@ def configure_web_proxy(
         request.install_opener(opener)
 
 
+def request_urllib(
+    req: request.Request, timeout: Optional[int] = None
+) -> HTTPResponse:
+    try:
+        resp = request.urlopen(req, timeout=timeout)
+    except error.HTTPError as e:
+        resp = e
+    except error.URLError as e:
+        raise exceptions.UrlError(e, url=req.full_url)
+
+    body = resp.read().decode("utf-8")
+
+    json_body = None
+    if "application/json" in resp.headers.get("Content-type", ""):
+        json_body = json.loads(body, cls=util.DatetimeAwareJSONDecoder)
+
+    return HTTPResponse(
+        code=resp.code,
+        headers=resp.headers,
+        body=body,
+        json=json_body,
```

Comment:
I am wonder if we should create a mechanism here to enforce that we expect a valid json response. If we add such mechanism, we would not need of the `isinstance` checks throught the code